doc_src/cmds/argparse.rst

```diff
@@ -163,6 +163,19 @@ The script should write any error messages to stdout, not stderr. It should retu
 
 Fish ships with a ``_validate_int`` function that accepts a ``--min`` and ``--max`` flag. Let's say your command accepts a ``-m`` or ``--max`` flag and the minimum allowable value is zero and the maximum is 5. You would define the option like this: ``m/max=!_validate_int --min 0 --max 5``. The default if you just call ``_validate_int`` without those flags is to simply check that the value is a valid integer with no limits on the min or max value allowed.
 
+Here are some examples of flag validations::
+
+  # validate that a path is a directory
+  argparse 'p/path=!test -d "$_flag_value"' -- --path $__fish_config_dir
+  # validate that a path does not exist or print an error
+  argparse 'p/path=!test ! -e "$_flag_value" || echo "Path exists: $_flag_value"' -- $argv
```

Comment:
This doesn't work. It'll print the error (to stdout), but it will still set $_flag_path.

That's because the `echo` exits true, so the entire script returns true.

You *could* abuse precedence:

```fish
argparse 'p/path=!test ! -e "$_flag_value"; or echo "Path exists: $_flag_value" && false' -- -p /etc
```

The `&&` binds to the `echo`, so this is two separate jobs:

```fish
test ! -e "$_flag_value"
```

or

```fish
echo "Path exists: $_flag_value" && false
```

However that's not something we want to encourage, so I think this should simply be left out here.

(there's a point to be made that a way to get argparse to easily error out if a flag fails validation, and to print a proper error message for it would be good. I don't know how that should work tho)

Reply:
> that's not something we want to encourage, so I think this should simply be left out here.

Good catch. I've removed that example and pushed a new commit with that example removed.

> there's a point to be made that a way to get argparse to easily error out if a flag fails validation, and to print a proper error message for it would be good. I don't know how that should work tho

One idea might have been better to `echo >&2 "Path exists: $_flag_value"` and then argparse could error if the validation writes anything to stderr. Or, something like `echo "Path exists: $_flag_value" && return 1`. That's a problem for another day though. No need to over complicate it - if the script is that long it's in _write a real validator function_ territory anyway.